cms/djangoapps/contentstore/middleware.py

```diff
@@ -0,0 +1,55 @@
+"""
+Middlewares for contentstore app.
+"""
+
+import logging
+
+from django.conf import settings
+from django.http import HttpResponse
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+
+from common.djangoapps.student.auth import (
+    STUDIO_EDIT_CONTENT,
+    STUDIO_VIEW_CONTENT,
+    get_user_permissions,
+)
+from openedx.core.lib.request_utils import COURSE_REGEX
+
+logger = logging.getLogger(__name__)
+
+
+class LimitedStaffAccessMiddleware:
+    """
+    Middleware to check access for limited staff users.
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def get_course_key(self, request):
+        """
+        Extract course key from request URI if it exists.
+        """
+        match = COURSE_REGEX.match(request.build_absolute_uri())
+        if match:
+            course_id = match.group("course_id")
+            try:
+                return CourseKey.from_string(course_id)
+            except InvalidKeyError:
+                logger.exception("Invalid course key: {}".format(course_id))
+        return None
+
+    def __call__(self, request):
+        user = request.user
+        course_key = self.get_course_key(request)
+
+        if settings.ROOT_URLCONF == "cms.urls" and course_key is not None:
```

Comment:
Do we need to check this? Wouldn't we only install this middleware in Studio?

Reply:
You're right since we only use middleware in `cms` we don't need to use such condition.